lesson24/index.html

```diff
@@ -0,0 +1,145 @@
+<!DOCTYPE html>
+<html lang="ja">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <meta http-equiv="X-UA-Compatible" content="ie=edge">
+        <link rel="stylesheet" href="./css/reset.css">
+        <link rel="stylesheet" href="./css/style.css">
+        <script src="./index.js" defer></script>
+        <title>Sign Up</title>
+    </head>
+    <body>
+        <form class="form box">
+            <h1 class="title">Sign Up</h1>
+            <dl class="form__inner">
+                <div class="form__item">
+                    <dt><label for="name">UserName</label></dt>
+                    <dd><input type="text" class="form__item-input" id="name" name="name"></dd>
+                </div>
+                <div class="form__item">
+                    <dt><label for="email">E-mail</label></dt>
+                    <dd><input type="email" class="form__item-input" id="email" name="email"></dd>
+                </div>
+                <div class="form__item">
+                    <dt><label for="password">Password</label></dt>
+                    <dd><input type="password" class="form__item-input" id="password" name="password"></dd>
+                </div>
+                <div class="form__check" id="js-checkbox-aria">
+                    <input type="checkbox" name="answer" title="answer" value="利用規約に同意しました" id="js-checkbox" disabled>
+                    <span class="form__check-text"><span class="form__check-link" id="js-checkbox-link">利用規約</span>に同意しました</span>
+                    <p class="form__attention">(規約を読むとチェックが入ります)</p>
+                </div>
+                <div class="form__button">
+                    <input type="button" id="js-submit-button" value="Create My Account">
+                </div>
+            </dl>
+        </form>
+        <div class="modal js-modal" id="js-modal-area">
+            <div class="modal__inner" id="js-modal-inner">
+                <button class="modal__close-button" id="js-modal-close-button" aria-label="Close"><img src="./img/icon-close.svg" alt="モーダルを閉じる" width="512" height="512" decoding=“async”></button>
+                <div class="modal__contents" id="js-modal-contents">
```

Comment:
IMO
I think this is `article`, since it stands alone as a sentence called "Terms of Use".
What do you think?
```suggestion
                <article class="modal__contents" id="js-modal-contents">
```

Reply:
Thanks so much for pointing that out!

Thanks so much for pointing that out!
I didn't know about the '<dialog>' element itself, so I got a big one.
I thought '<dialog>' would be suitable, but I was told that older versions may not work, so I took a big risk and added a role attribute to the div.
[My referenced site](https://www.w3.org/WAI/GL/wiki/Using_ARIA_role%3Ddialog_to_implement_a_modal_dialog_box)

[4840a64](https://github.com/haru-programming/JavaScript-lessons/pull/28/commits/4840a6473f223c15d6b622df5c6e60f5705cd618)